submissions/maxim-zabolotny/friends-app/app.js

```diff
@@ -0,0 +1,215 @@
+const APP_CONFIG = {
+    url: 'https://randomuser.me/api/?results=60',
+    filterBy: 'name'
+}
+const CSS_CLASSES = {
+    card: 'card',
+    radioButton: 'radio-box',
+    navButton: 'nav-button'
+}
+
+const DOM_NODES = {
+    inputText: document.querySelector('.input-text'),
+    radioButtons: document.querySelectorAll('.radio-box'),
+    form: document.querySelector('.filters'),
+    target: document.querySelector('.frame'),
+    btnWrapper: document.querySelector('.btn-wrapper'),
+    resetBtn: document.querySelector('.filter .reset_button'),
+}
+
+const state = {
+    allUsers : [],
+    filteredUsers: [],
+    sortedUsers: []
+}
+
+window.state = state
+
+const SORT = {
+    all: () => renderCards(state.sortedUsers = [...state.filteredUsers]),
+    female: () => renderCards(state.sortedUsers = state.filteredUsers.filter(user => user.gender.toLowerCase() == 'female')),
+    male: () => renderCards(state.sortedUsers = state.filteredUsers.filter(user => user.gender.toLowerCase() == 'male')),
+    nameAsc: () => renderCards(state.sortedUsers = state.filteredUsers.sort((a, b) => {
+        if(a.name.first.toLowerCase() > b.name.first.toLowerCase()) return 1
+        else return -1
+    })),
+    nameDesc: () => renderCards(state.sortedUsers = state.filteredUsers.sort((a, b) => {
+        if(a.name.first.toLowerCase() < b.name.first.toLowerCase()) return 1
+        else return -1
+    })),
+    ageAsc: () => renderCards(state.sortedUsers = state.filteredUsers.sort((a, b) => {
+        if(a.age > b.age) return 1
+        else return -1
+    })),
+    ageDesc: () => renderCards(state.sortedUsers = state.filteredUsers.sort((a, b) => {
+        if(a.age < b.age) return 1
+        else return -1
+    })),
+}
+
+const capitalized = ({first, last}) => {
+    let str = `${first} ${last}`
```

Comment:
Maybe it would be better to uppercase the first name and the last name directly. I don't see a reason to combine them in one string just to split it back in the next line. Am I missing something?

Reply:
This function is used to process the variant in which the first and last name with a small letter comes from the server

Reply:
BTW I would change function name `getCapitalizedName`, because  `capitalized` sounds more like a value, than a function (function usually does something, so generally it should have a verb in it's name)

What I mean is that you create a string `str` only to split it in the next line. How about something like that:
```javascript
const capitalize = str => str.charAt(0).toUpperCase() + str.slice(1);

const getCapitalizedName  = ({first, last}) => {
    return `${capitalize(first)} ${capitalize(last)}`
}
```

Reply:
Thank you. I understand what you mean